spotbugsTestCases/src/java11/Issue2182.java

```diff
@@ -0,0 +1,27 @@
+package ghIssues;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class Issue2182 {
+
+    public void test() {
+        List<String> A = new ArrayList<String>();
+        A.add("Alex");
+        A.add("john");
+        A.add("lily");
+        A.add("tracy");
+        Iterator <String>it = A.iterator();
+
+        it = A.iterator();
+        String add = "";
+        while (it.hasNext()) {
+            String retrieve = it.next();
+            if (retrieve != null && !"".equals(retrieve))
```

Comment:
It's better to always use braces with `if`, so that you never [goto fail](https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/).

Reply:
I agree. This was just copied over from an existing test.